### PR TITLE
Feat: velaql provider for get application resource-tree

### LINKED ISSUE #3886: [Feature] Proposal: show the resource-tree of application
**Is your feature request related to a problem? Please describe.**

We should show the resource-tree of an application. Here are some principles of the requirement.

1. show all managed resources of an application and their children-resources
2. show some basic status of the resource such as  `running or pending` of a pod
3. must support in both velaux and cli
4. resources in all cluster should be considerd.

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by wangyikewxgm**
implementation for get application resourceTree with velaql

eg: When access the endpoint: http://127.0.0.1:8000/api/v1/query/?velaql=application-resource-tree-view{appName="addon-kruise-rollout",appNs="vela-system",}.status . The result is
```json
{
    "resources": [
        {
            "apiVersion": "source.toolkit.fluxcd.io/v1beta1",
            "cluster": "",
            "component": "kruise-rollout",
            "kind": "HelmRepository",
            "latest": true,
            "name": "kruise-rollout",
            "namespace": "vela-system",
            "resourceTree": {
                "apiVersion": "source.toolkit.fluxcd.io/v1beta1",
                "cluster": "",
                "kind": "HelmRepository",
                "name": "kruise-rollout",
                "namespace": "vela-system",
                "resourceHealthStatus": {
                    "message": "",
                    "reason": "",
                    "statusCode": "HealthStatusHealthy"
                }
            },
            "revision": "addon-kruise-rollout-v1",
            "trait": ""
        },
        {
            "apiVersion": "helm.toolkit.fluxcd.io/v2beta1",
            "cluster": "",
            "component": "kruise-rollout",
            "kind": "HelmRelease",
            "latest": true,
            "name": "kruise-rollout",
            "namespace": "vela-system",
            "resourceTree": {
                "apiVersion": "helm.toolkit.fluxcd.io/v2beta1",
                "childrenResources": [
                    {
                        "apiVersion": "v1",
                        "cluster": "",
                        "kind": "Service",
                        "name": "kruise-rollout-webhook-service",
                        "namespace": "kruise-rollout",
                        "resourceHealthStatus": {
                            "message": "",
                            "reason": "",
                            "statusCode": "HealthStatusHealthy"
                        },
                        "uid": "a80aa2bc-92aa-48a1-bd30-db88a53218b4"
                    },
                    {
                        "apiVersion": "apps/v1",
                        "childrenResources": [
                            {
                                "apiVersion": "apps/v1",
                                "childrenResources": [
                                    {
                                        "apiVersion": "v1",
                                        "cluster": "",
                                        "kind": "Pod",
                                        "name": "kruise-rollout-controller-manager-696bdb76f8-626cv",
                                        "namespace": "kruise-rollout",
                                        "resourceHealthStatus": {
                                            "message": "",
                                            "reason": "",
                                            "statusCode": "HealthStatusHealthy"
                                        },
                                        "uid": "3f11c45f-cf5b-4e20-bad2-9f4bcaccead9"
                                    },
                                    {
                                        "apiVersion": "v1",
                                        "cluster": "",
                                        "kind": "Pod",
                                        "name": "kruise-rollout-controller-manager-696bdb76f8-7g4jj",
                                        "namespace": "kruise-rollout",
                                        "resourceHealthStatus": {
                                            "message": "",
                                            "reason": "",
                                            "statusCode": "HealthStatusHealthy"
                                        },
                                        "uid": "6b942769-ae38-4751-863e-2c99bf0929a7"
                                    }
                                ],
                                "cluster": "",
                                "kind": "ReplicaSet",
                                "name": "kruise-rollout-controller-manager-696bdb76f8",
                                "namespace": "kruise-rollout",
                                "resourceHealthStatus": {
                                    "message": "",
                                    "reason": "",
                                    "statusCode": "HealthStatusHealthy"
                                },
                                "uid": "96414447-a222-4301-a122-86f417d9af48"
                            }
                        ],
                        "cluster": "",
                        "kind": "Deployment",
                        "name": "kruise-rollout-controller-manager",
                        "namespace": "kruise-rollout",
                        "resourceHealthStatus": {
                            "message": "",
                            "reason": "",
                            "statusCode": "HealthStatusHealthy"
                        },
                        "uid": "a0935033-d0b6-4b32-a20f-59a60ee5ded4"
                    }
                ],
                "cluster": "",
                "kind": "HelmRelease",
                "name": "kruise-rollout",
                "namespace": "vela-system",
                "resourceHealthStatus": {
                    "message": "",
                    "reason": "",
                    "statusCode": "HealthStatusHealthy"
                }
            },
            "revision": "addon-kruise-rollout-v1",
            "trait": "AuxiliaryWorkload"
        }
    ]
}
```

### Description of your changes

Fixes #3886 

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

**Review by Somefive — APPROVED** [review]

File: pkg/stdlib/pkgs/query.cue [hunk 1]
```diff
@@ -124,3 +124,29 @@
 	}]
 	...
 }
+
+#GetApplicationTree: {
+	#do:       "getApplicationTree"
+	#provider: "query"
+	app: {
+		name:      string
+		namespace: string
```
Comment (barnettZQG) [inline pkg/stdlib/pkgs/query.cue:133]: miss the filter option?

Comment (wangyikewxgm) [reply pkg/stdlib/pkgs/query.cue:133]: I don't get the scenario. Why we need the filter when show a application tree?

File: pkg/velaql/providers/query/types/type.go [hunk 2]
```diff
@@ -87,19 +87,32 @@ type Endpoint struct {
...
+
+// ResourceTreeNode is the tree node of every resource
+type ResourceTreeNode struct {
+	Cluster              string             `json:"cluster"`
+	APIVersion           string             `json:"apiVersion,omitempty"`
+	Kind                 string             `json:"kind"`
+	Namespace            string             `json:"namespace,omitempty"`
+	Name                 string             `json:"name,omitempty"`
+	UID                  types.UID          `json:"uid,omitempty"`
+	ResourceHealthStatus HealthStatus       `json:"resourceHealthStatus,omitempty"`
```
Comment (wonderflow) [inline pkg/velaql/providers/query/types/type.go:114]:
```suggestion
	ResourceHealthStatus HealthStatus       `json:"healthStatus,omitempty"`
```

File: pkg/velaql/providers/query/types/type.go [hunk 3]
```diff
@@ -87,19 +87,32 @@ type Endpoint struct {
...
+	ChildrenResources    []ResourceTreeNode `json:"childrenResources,omitempty"`
```
Comment (wonderflow) [inline pkg/velaql/providers/query/types/type.go:115]:
```suggestion
	ChildrenResources    []ResourceTreeNode `json:"leafNodes,omitempty"`
```

File: pkg/velaql/providers/query/tree.go [hunk 4]
```diff
@@ -0,0 +1,439 @@
...
+		}
+	}
+	err = k8sClient.List(clusterCTX, &itemList, &listOptions)
+	if err != nil {
+		return nil, err
+	}
+	return itemList.Items, nil
+}
+
+func iteratorChildResources(ctx context.Context, cluster string, k8sClient client.Client, parentResource types.ResourceTreeNode) ([]types.ResourceTreeNode, error) {
```
Comment (wonderflow) [inline pkg/velaql/providers/query/tree.go:396]: we should add some limit for the depth, I'm afraid there will be potential performance issues.

We can add a default max-depth to 5 or 10, if you don't have any recommend value. You can check some typical scenarios and find one.

Comment (wangyikewxgm) [reply pkg/velaql/providers/query/tree.go:396]: In my tests, most scenario cannot reach 5, helmRelease -> deployment -> rs -> pod

Comment (wonderflow) [reply pkg/velaql/providers/query/tree.go:396]: Then 5 is safe here.

File: pkg/velaql/providers/query/tree.go [hunk 5]
```diff
@@ -393,7 +396,10 @@ func listItemByRule(clusterCTX context.Context, k8sClient client.Client, resourc
 	return itemList.Items, nil
 }
 
-func iteratorChildResources(ctx context.Context, cluster string, k8sClient client.Client, parentResource types.ResourceTreeNode) ([]types.ResourceTreeNode, error) {
+func iteratorChildResources(ctx context.Context, cluster string, k8sClient client.Client, parentResource types.ResourceTreeNode, depth int) ([]types.ResourceTreeNode, error) {
+	if depth == maxDepth {
```
Comment (wonderflow) [inline pkg/velaql/providers/query/tree.go:400]:
```suggestion
	if depth > maxDepth {
```

Shall we return some error or logs for reminder?

Comment (wangyikewxgm) [reply pkg/velaql/providers/query/tree.go:400]: I think we cannot generate any error, an error will break whole return chain, the result of this velaql will return error.

**Review by wonderflow — APPROVED** [review]

File: charts/vela-core/templates/velaql/resourceTree.yaml [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+kind:       "ConfigMap"
+metadata:
+  name:      "application-resource-tree-view"
+  namespace: {{ include "systemDefinitionNamespace" . }}
+data:
+  template: |
+    import (
+        "vela/ql"
+    )
+    parameter: {
```
Comment (barnettZQG) [inline charts/vela-core/templates/velaql/resourceTree.yaml:11]: missing the name、cluster and clusterNs fields.

Comment (wangyikewxgm) [reply charts/vela-core/templates/velaql/resourceTree.yaml:11]: good catch, thanks.

**Review by barnettZQG — APPROVED** [review]

**Review by leejanee — APPROVED** [review]

**Pull request merged**